app/javascript/packs/containers/QuestionnaireForm/QuestionnaireFormContainer.jsx

```diff
@@ -1,69 +1,108 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { getQuestionsByBuildingType } from '../../selectors/questionsSelector';
+import { getCategoryByBuildingTypeId } from '../../selectors/categoriesSelector';
 import { getBuildingType } from '../../selectors/buildingTypesSelector';
-import QuestionContainer from './QuestionContainer';
-import { beforeCreateNewQuestion } from '../../actions/questions';
-import { generateTempId } from '../../utils/TemporaryObjectUtil';
 import PropTypes from 'prop-types';
+import CategoryQuestionsContainer from './CategoryQuestionsContainer';
+import {generateTempId} from '../../utils/TemporaryObjectUtil';
+import {patch, post} from '../../fetch/requester';
+import {
+  categoryFetchFailure, categoryFetchInProgress, categoryFetchSuccess, categoryPreFetchSave,
+  removeCategory,categorySetNew
+} from '../../actions/categories';
+
+import CategoryDisplay from '../../components/QuestionnaireForm/CategoryDisplay';
 
 class QuestionnaireFormContainer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      currentCategory: this.props.categories[0],
+      select: false
+    };
+  }
 
-  /**
-   * Handles creating a new temp question when Add Question button is clicked.
-   */
-  onNewQuestion() {
-    const newQuestion = {
+  toggleCategory(category) {
+    this.setState({currentCategory: category})
+  }
+
+  onNewCategory() {
+    const newCategory = {
       id: generateTempId(),
-      text: "",
+      name: "Untitled",
       building_type_id: this.props.buildingType.id,
-      category_id: 1,
-      options: {},
-      question_type: null,
-      parameter: null,
-      helper_text: null
     };
-    this.props.beforeCreateNewQuestion(newQuestion)
+    this.createCategory(newCategory)
+  }
+
+  /**
+   * Handles fetch request to post a question and redux update
+   * @param { string } id - questionId to update
+   * @param { Object } args - any question parameters
+   */
+  async createCategory(newCategory) {
+    try {
+      let response = await post('/api/categories', {'category': newCategory});
+      this.props.categoryFetchSuccess(response.data);
+      this.props.categorySetNew(response.data.id);
+      this.setState({currentCategory: response.data})
+    } catch (error) {
+      this.props.categoryFetchFailure(error, newCategory);
+    }
   }
 
   render() {
-    const questions_display = Object.keys(this.props.questions).map((id)=>{
-      const question = this.props.questions[id];
-      if (!question.parent_option_id) {
-        return(
-          <div key={id}>
-            <QuestionContainer
-              question={question}
-            />
-          </div>
-        )
-      }
+    const categoryToggle = Object.keys(this.props.categories).map((categoryId) => {
```

Comment:
`this.props.categories` returns an array of categories, which means the indices of this map will _not_ be `categoryId`s but indices!